lib/ace/ext/papeeria/highlighter.coffee

```diff
@@ -1,140 +1,192 @@
+			positionLeftwards.row += 1
+			positionLeftwards.column = 0
+		else
+			positionLeftwards.column += 1
+
+		positionRightwards = editor.getCursorPosition()        
+
+		allBrackets =
+			left: [
+				session.$findOpeningBracket('}', positionLeftwards, /(\.?.paren)+/)
+				session.$findOpeningBracket(']', positionLeftwards, /(\.?.paren)+/)
+				session.$findOpeningBracket(')', positionLeftwards, /(\.?.paren)+/)
+			]
+			right: [
+				session.$findClosingBracket('{', positionRightwards, /(\.?.paren)+/)
+				session.$findClosingBracket('[', positionRightwards, /(\.?.paren)+/)
+				session.$findClosingBracket('(', positionRightwards, /(\.?.paren)+/)
+			]
+		leftNearest = null
+		rightNearest = null
+		key = 0
+		while key < allBrackets.left.length
+			leftCandidate = allBrackets.left[key]
+			rightCandidate = allBrackets.right[key]
+			if !leftNearest
+				leftNearest = leftCandidate
+			if !rightNearest
+				rightNearest = rightCandidate
+			if leftCandidate
+				if leftNearest.row <= leftCandidate.row
+					if leftNearest.row == leftCandidate.row
+						if leftNearest.column < leftCandidate.column
+							leftNearest = leftCandidate
+					else
+						leftNearest = leftCandidate
+			if rightCandidate
+				if rightNearest.row >= rightCandidate.row
+					if rightNearest.row == rightCandidate.row
+						if rightNearest.column > rightCandidate.column
+							rightNearest = rightCandidate
+					else
+						rightNearest = rightCandidate
+			key++
+
+		result =
+			left: leftNearest
+			right: rightNearest
+			mismatch: true
+			equals: (object) ->
+
+				if object.left != @left
+					return false
+				if object.right != @right
+					return false
+				if object.mismatch != @mismatch
+					return false
+				return true
+		
+		if result.left && result.right
+			expectedRightBracket = session.$brackets[session.getLine(result.left.row).charAt(result.left.column)]
+			rightBracket = session.getLine(result.right.row).charAt(result.right.column)
+			if  expectedRightBracket == rightBracket
+				result.mismatch = false
+		return result
+
+
+	popoverHandler = popoverHandler ? {
+		options: {
+			html: true
+			placement: "bottom"
+			trigger: "manual"
+			container: "#editor"
+		}
+
+		show: (jqPopoverContainer, content) ->
+			setTimeout(->
+				cursorPosition = $("textarea.ace_text-input").position()
+				jqPopoverContainer.css({
+				  top: cursorPosition.top + 24 + "px"
+				  left: cursorPosition.left + "px"
+				})
+				popoverHandler.options.content = content
+				jqPopoverContainer.popover(popoverHandler.options)
+				jqPopoverContainer.popover("show")
+				return
+			, 100)
+
+		hide: (jqPopoverContainer) ->
+			jqPopoverContainer.popover("destroy")
+	}
+
+	initPopover = (editor) ->
+		cssPath = require.toUrl("./highlighter.css")
+		linkDemo = $("<link>").attr(
+			rel: "stylesheet"
+			href: cssPath
+		)
+		$("head").append(linkDemo)
+
+		span = $("<span>").attr(
+			id: "line"
+		)
+
+		$("body").append(span)
```

Comment:
For demo purposes you can just change the markup and add popover container there.